src/snapred/backend/recipe/algorithm/LiteDataCreationAlgo.py

```diff
@@ -0,0 +1,99 @@
+import json
+
+from mantid.api import *
+from mantid.api import AlgorithmFactory, PythonAlgorithm, mtd
+from mantid.kernel import Direction
+
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+
+name = "LiteDataCreationAlgo"
+
+
+class LiteDataCreationAlgo(PythonAlgorithm):
+    def PyInit(self):
+        self.declareProperty("InputWorkspace", defaultValue="", direction=Direction.Input)
+        self.declareProperty("OutputWorkspace", defaultValue="", direction=Direction.Output)
+        self.setRethrows(True)
+        self.mantidSnapper = MantidSnapper(self, name)
+
+    def PyExec(self):
+        self.log().notice("Lite Data Creation START!")
+        # define super pixel dimensions
+        superPixel = [8, 8]
+
+        # get output workspace name
+        outputWorkspaceName = self.getProperty("OutputWorkspace").value
+
+        # load input workspace
+        inputWorkspace = self.getProperty("InputWorkspace").value
+        self.mantidSnapper.mtd[inputWorkspace]
+
+        # clone the workspace
+        self.mantidSnapper.CloneWorkspace(
+            "Cloning input workspace for lite data creation...",
+            InputWorkspace=inputWorkspace,
+            OutputWorkspace=outputWorkspaceName,
+        )
+        self.mantidSnapper.executeQueue()
+        outputWorkspace = self.mantidSnapper.mtd[outputWorkspaceName]
+
+        # IEventWorkspace ID is equivalent to spec ID
+        numSpec = outputWorkspace.getNumberHistograms()
+
+        # array to hold spectrum data
+        spectrumData = []
+
+        # iterate over IEventWorkspace IDs
+        for spec in range(numSpec):
+            # get event list per workspace
+            spectrum = outputWorkspace.getSpectrum(workspaceIndex=spec)
+            spectrumData.append(spectrum)
+
+        # allocate array of event IDs
+        for spectrum in spectrumData:
+            eventIDs = spectrum.getDetectorIDs()
+            for event in eventIDs:
+                superID = self.getSuperID(nativeID=event, xdim=superPixel[0], ydim=superPixel[0])
+
+                # clear detector IDs and then add the new modified super ID
+                spectrum.clearDetectorIDs()
+                spectrum.addDetectorID(superID)
+
+        self.mantidSnapper.LoadInstrument(
+            "Loading instrument...",
+            Workspace=outputWorkspace,
+            Filename="/SNS/SNAP/shared/Calibration/Powder/SNAPLite.xml",
```

Comment:
I'm worried this ties this algorithm to only be able to run on analysis.  Is there a way to specify this filename, for instance to point at `tests/resources/inputs/calibration/fakeSNAPLite.xml`?

Reply:
The purpose of the algo is to run on the analysis machine. I understand what you are getting at but it needs to implement a new instrument definition. Are all the values contained within fakeSNAPLite.xml the same as the ones within SNAPLite.xml? In that case, this change would be fine.

Reply:
After examination, these instrument definition files are not the same. The one within the inputs directory is explicitly for testing purposes only as seen in the comments. There are other details included as well within SNAPLite.xml that are missing within fakeSNAPLite.xml.

Reply:
there is a SNAPLite.xml defined inside the test resources

Reply:
I've changed the path to /tests/resources/inputs/pixel_grouping/SNAPLite_Definition.xml